.github/workflows/cnvm-ci.yml

```diff
@@ -0,0 +1,75 @@
+name: CNVM-CI
+permissions:
+  id-token: write
+  contents: read
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+"
+    types: [opened, synchronize, reopened]
+
+jobs:
+  Run-CNVM-Tests:
+    name: CNVM integration test
+    runs-on: ubuntu-22.04
+    timeout-minutes: 40
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Init Hermit
```

Comment:
Do we actually need hermit? It seems that everything is initialized with actions

Reply:
It is preferable we init with hermit instead of actions, to make the actions flow as close as possible to our dev environment. Basically to avoid cases where it works in the CI but fails locally

Reply:
I still don't get why we need to init python here separately (also the action takes 0 seconds so I'm not sure what it's achieving)

Reply:
Agreed. In general, it seems that we have many workflows that need to repeat the same steps. Should we try to centralize the setup and share it across different jobs?

Reply:
We switch to python3.9 for installing `poetry` that for some reason can't be installed with the python version that provided by hermit.

Reply:
what errors are you getting? did it work locally?

Reply:
> Agreed. In general, it seems that we have many workflows that need to repeat the same steps. Should we try to centralize the setup and share it across different jobs?

which steps? init hermit?

Reply:
> > Agreed. In general, it seems that we have many workflows that need to repeat the same steps. Should we try to centralize the setup and share it across different jobs?
> 
> @orestisfl which steps? init hermit?

hermit, setup go, build cloudbeat. In other pipelines there are also some duplicated docker image builds.